### PR TITLE
Make external links more obvious

### LINKED ISSUE #83: Improve External Links
As a user involved in the governance, risk, and compliance of systems and components, I want it to be obvious which links in the OSCAL viewer are external so that I can better determine whether to explore that link during my workflow at that time.

### Acceptance Criteria
* All external links display the OpenInNew material-ui icon
* All external links open in a new browser tab

### PR DESCRIPTION
**Pull request by tuckerzp**
Closes #83

### CODE REVIEW

File: example/src/App.css [hunk 1]
```diff
@@ -36,3 +36,9 @@
     transform: rotate(360deg);
   }
 }
+
+a[href^="http"]:not([data-internal-link]):before {
+  margin-left: 10px;
+  content: '[external]';
```
Comment (tuckerzp) [inline example/src/App.css:42]: I do not see an obvious way to include [this icon](https://materialui.co/icon/open-in-new) in css.

Comment (kylelaker) [reply example/src/App.css:42]: What about not doing a CSS-only solution? Can we create a wrapper around `Link` that behaves differently based on the start of `href`?

https://mui.com/material-ui/react-link/

This is really bad and stupid and probably doesn't work. But maybe it conveys my thoughts?

```tsx
import * as React from 'react';
import Link as MuiLink from '@mui/material/Link';
import OpenInNewIcon from '@mui/icons-material/OpenInNew';

export default function Link(props) {
  if (!props.href.startsWith('http')) {
    return (<MuiLink {...props}/>);
  }
  return (
    <MuiLink
      {...props}
      target="_blank"
      rel="noopener"
    >
      <OpenInNewIcon/>{children}
    </MuiLink>
  );
}
```

I think https://github.com/remarkjs/react-markdown#appendix-b-components can be used with our new Markdown component to remap `<a>` tags to use this too.

**Review by Bronstrom — APPROVED** [review]

**Review by kylelaker — CHANGES_REQUESTED** [review]
This will only impact links in the Back Matter. What about external links that may appear elsewhere such as in `markup-multiline` typed fields?

File: src/components/OSCALMarkdownProse.js [hunk 2]
```diff
@@ -1,12 +1,23 @@
 import React from "react";
 import ReactMarkdown from "react-markdown";
+import OpenInNewIcon from "@mui/icons-material/OpenInNew";
+
+const components = {
```
Comment (tuckerzp) [inline src/components/OSCALMarkdownProse.js:5]: `react-markdown` allows for you to change things that come from markdown. 

https://github.com/remarkjs/react-markdown#appendix-b-components

We rewrite the `<a>` tag to display the external link icon.

**Review by Bronstrom — CHANGES_REQUESTED** [review]
It looks like there is also an external link for Categorization ID in `OSCALSystemCharacteristics`

**Comment by tuckerzp** [comment]
To make the inevitable conflict a bit easier to deal with, tests will fail until #548 is in. It should be a simple change to get them to pass again.

File: src/components/OSCALMarkupProse.js [hunk 3]
```diff
@@ -1,32 +1,35 @@
...
+};
+
 /**
  * Renders a string of markdown to React elements
  * @param {String} props.text
  * @returns a React element from the markdown
  */
 
 export function OSCALMarkupMultiLine(props) {
+  components.p = props.paragraphComponent ?? "p";
```
Comment (kylelaker) [inline src/components/OSCALMarkupProse.js:24]: This changes the base `components` object in a way that I am not especially comfortable with.

```tsx
<ReactMarkdown
  components = {{
    ...components,
    p: props.paragraphComponent ?? "p",
  }}
>
  {props.children}
</ReactMarkdown> 
```

File: src/components/OSCALMarkupProse.js [hunk 4]
```diff
@@ -1,32 +1,35 @@
...
+const components = {
+  a: ({ ...props }) =>
+    props.href.startsWith("http") ? (
+      <a target="_blank" {...props}>
+        <OpenInNewIcon /> {props.children}
+      </a>
+    ) : (
+      <a {...props}>{props.children}</a>
+    ),
+};
```
Comment (kylelaker) [inline src/components/OSCALMarkupProse.js:15]: I'd like to name this something a little more descriptive like `baseComponents` or something like that.

File: src/components/OSCALMarkupProse.js [hunk 5]
```diff
@@ -1,32 +1,35 @@
 import React from "react";
 import ReactMarkdown from "react-markdown";
 
+import OpenInNewIcon from "@mui/icons-material/OpenInNew";
+
+const components = {
+  a: ({ ...props }) =>
+    props.href.startsWith("http") ? (
+      <a target="_blank" {...props}>
```
Comment (kylelaker) [inline src/components/OSCALMarkupProse.js:9]: Why `<a>` and not `<Link>` from MUI?

**Review by kylelaker — APPROVED** [review]

**Review by Bronstrom — APPROVED** [review]

**Pull request merged**